src/goto-symex/goto_symex.cpp

```diff
@@ -118,8 +118,9 @@ void goto_symext::symex_assign(
     }
 
     exprt::operandst lhs_if_then_else_conditions;
-    symex_assign.assign_rec(
-      lhs, expr_skeletont{}, rhs, lhs_if_then_else_conditions);
+    symex_assignt{
+      shadow_memory, state, assignment_type, ns, symex_config, target}
```

Comment:
Should `shadow_memory` perhaps become part of `state` (`goto_symex_statet`)?

Reply:
It may be possible to do something similar to `guard_manager` which is also passed through goto_symex_statet. That would leave the symex_assignt constructor untouched, but add a parameter to goto_symex_statet.